airflow-core/docs/core-concepts/executor/index.rst

```diff
@@ -251,7 +252,7 @@ Mandatory Methods to Implement
 The following methods must be overridden at minimum to have your executor supported by Airflow:
 
 * ``sync``: Sync will get called periodically during executor heartbeats. Implement this method to update the state of the tasks which the executor knows about. Optionally, attempting to execute queued tasks that have been received from the scheduler.
-* ``execute_async``: Executes a command asynchronously. A command in this context is an Airflow CLI command to run an Airflow task. This method is called (after a few layers) during executor heartbeat which is run periodically by the scheduler. In practice, this method often just enqueues tasks into an internal or external queue of tasks to be run (e.g. ``KubernetesExecutor``). But can also execute the tasks directly as well (e.g. ``LocalExecutor``). This will depend on the executor.
+* ``execute_async``: Executes a command (Airflow 2) /workload(Airflow 3) asynchronously. A command is an Airflow CLI command whereas workload is basic unit of work that represents an Airflow task. This method is called (after a few layers) during executor heartbeat which is run periodically by the scheduler. In practice, this method often just enqueues tasks into an internal or external queue of tasks to be run (e.g. ``KubernetesExecutor``). But can also execute the tasks directly as well (e.g. ``LocalExecutor``). This will depend on the executor.
```

Comment:
I think we need to refine from doing the AF2 / AF3 like that.
It would be a challenging for users to find what they need and for us to maintain the docs.
For example this raise the question what is command and what is workload.

Note that the word workload is not defined in the doc.

I suggest the docs should speak only AF3 language like from a perspective.
In the cases where we want to explain something in AF2 vs AF3 we should do it in a designated paragraph.

Reply:
Noting that the docs are versioned. so if someone want to see the content of this page in Airflow 2 they can just click the version button

Reply:
Actually that makes sense, let me reword this thing.

Reply:
Honestly, I'm not sure this paragraph needs to even say whether it's a cli command or workload. But I like Elad's suggestion if it must.

Reply:
Checked with Elad and did a re work on this, he was ok with it